framework/validators/IpAddressValidator.php

```diff
@@ -0,0 +1,432 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * IpAddressValidator validates that the attribute value is a valid IPv4/IPv6 address or subnet.
+ * May change attribute's value if normalization is enabled.
+ *
+ * @author SilverFire <d.naumenko.a@gmail.com>
+ * @since 2.0.4
+ */
+class IpAddressValidator extends Validator
+{
+    /**
+     * @const integer the length of IPv6 address in bits
+     */
+    const IPV6_ADDRESS_LENGTH = 128;
+
+    /**
+     * @const integer the length of IPv4 address in bits
+     */
+    const IPV4_ADDRESS_LENGTH = 32;
+
+    const RANGE_ORDER_ALLOWED_DENIED = 0;
+    const RANGE_ORDER_DENIED_ALLOWED = 1;
+
+    /**
+     * @var boolean whether support of IPv6 addresses is enabled
+     */
+    public $ipv6 = true;
+
+    /**
+     * @var boolean whether support of IPv4 addresses is enabled
+     */
+    public $ipv4 = true;
+
+    /**
+     * @var boolean|string whether address can be a CIDR subnet
+     *     boolean - normal behaviour
+     *     string  - value 'only' to validate only the addresses with a CIDR
+     */
+    public $subnet = false;
+
+    /**
+     * @var boolean whether to add the prefix with the smallest length (32 for IPv4 and 128 for IPv6)
+     * to an address without it.
+     * Works only when attribute [[subnet]] is not false.
+     * @see subnet
+     */
+    public $normalize = false;
+
+    /**
+     * @var boolean|string whether address may have an exclude-character at the beginning
+     *   boolean - character "!" will be used
+     *   string - passed character will be used
+     */
+    public $exclude = false;
```

Comment:
I don't think that we should drop the `exclude` property. It is not always necessary to allow the `!` character at the beginning of the IP address. If the system does not expect the use of the exception character at the beginning of the IP address, this can result in some abnormal behavior or even in vulnerability.